src/main/scala/chisel3/util/experimental/minimizer/QMCMinimizer.scala

```diff
@@ -0,0 +1,121 @@
+package chisel3.util.experimental.minimizer
+
+import chisel3.util.BitPat
+
+object QMCMinimizer {
+  /** @return A instance of [[QMCMinimizer]] */
+  def apply(): QMCMinimizer = new QMCMinimizer()
+
+  implicit class Implicant(x: BitPat) {
+    /** Check whether two implicants have the same value on all of the cared bits (intersection).
+      *
+      * {{{
+      * value ^^ x.value                                       // bits that are different
+      * (bits that are different) & x.mask                     // bits that are different and `this` care
+      * (bits that are different and `this` care) & y.mask     // bits that are different and `both` care
+      * (bits that are different and both care) == 0           // no (bits that are different and we both care) exists
+      * no (bits that are different and we both care) exists   // all cared bits are the same, two terms intersect
+      * }}}
+      *
+      * @param y Implicant to be checked with
+      * @return Whether two implicants intersect
+      */
+    def intersects(y: BitPat): Boolean = ((x.value ^ y.value) & x.mask & y.mask) == 0
+
+    /** Check if two implicants are the same
+      * @param y Implicant to be checked with
+      * @return Whether two implicants are the same
+      */
+    def sameAs(y: BitPat): Boolean = x.mask == y.mask && x.value == y.value
+
+    /** Merge two similar implicants if they are similar
+      * Rule of merging: '0' and '1' merge to '?'
+      * Two implicants are "similar" when they satisfy all the following rules:
+      *   1. have the same mask ('?'s are at the same positions)
+      *   1. values only differ by one bit
+      *
+      * @example this = 11?0, x = 10?0 -> similar
+      * @example this = 11??, x = 10?0 -> not similar, violated rule 1
+      * @example this = 11?1, x = 10?0 -> not similar, violated rule 2
+      * @param y Implicant to be merged with
+      * @return Merge result wrapped in [[Some]] if `x` and `y` are similar,
+      *         [[None]] otherwise
+      */
+    def mergeIfSimilar(y: BitPat): Option[BitPat] = {
+      val diff = x.value ^ y.value
+      if (x.mask == y.mask && diff.bitCount == 1) { // similar
```

Comment:
first instinct: `x.mask == y.mask` is too strict which might loss some corner cases, need to be discussed later.

Reply:
`0` + `1` -> `?`